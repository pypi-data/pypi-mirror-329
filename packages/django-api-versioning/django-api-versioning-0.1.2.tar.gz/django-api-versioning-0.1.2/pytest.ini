[pytest]
DJANGO_SETTINGS_MODULE = django_api_versioning.settings
python_files = tests.py test_*.py *_tests.py
addopts = --cov=django_api_versioning --cov-report=term-missing