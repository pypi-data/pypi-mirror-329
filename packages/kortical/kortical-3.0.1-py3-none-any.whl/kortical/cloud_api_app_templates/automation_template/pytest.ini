[pytest]
markers =
    unit: runs tests that don't need an instantiated system.
    integration: runs tests that do require an instantiated system
    smoke: run the minimum tests to test core functionality