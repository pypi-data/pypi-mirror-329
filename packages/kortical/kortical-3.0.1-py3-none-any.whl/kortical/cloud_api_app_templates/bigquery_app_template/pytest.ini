[pytest]
markers =
    api: runs api endpoint tests.