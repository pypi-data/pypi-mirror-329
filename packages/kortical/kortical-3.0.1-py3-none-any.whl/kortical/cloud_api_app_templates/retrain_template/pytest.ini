[pytest]
markers =
    api: runs api endpoint tests.
    smoke: runs a faster set of tests to validate functionality.