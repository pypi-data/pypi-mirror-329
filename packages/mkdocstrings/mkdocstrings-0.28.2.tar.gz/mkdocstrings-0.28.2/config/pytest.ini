[pytest]
python_files =
  test_*.py
addopts =
  --cov
  --cov-config config/coverage.ini
testpaths =
  tests

# action:message_regex:warning_class:module_regex:line
filterwarnings =
  error
  # TODO: remove once pytest-xdist 4 is released
  ignore:.*rsyncdir:DeprecationWarning:xdist
  # TODO: remove once mkdocstrings-python releases a new version
  ignore:.*`handler` argument:DeprecationWarning:mkdocstrings_handlers
  ignore:.*`mdx` argument:DeprecationWarning:mkdocstrings_handlers
  ignore:.*`mdx_config` argument:DeprecationWarning:mkdocstrings_handlers
  ignore:.*`update_env\(md\)` parameter:DeprecationWarning:mkdocstrings
  ignore:.*`super\(\).update_env\(\)` anymore:DeprecationWarning:mkdocstrings_handlers
  ignore:.*`get_anchors` method:DeprecationWarning:mkdocstrings
  ignore:.*fallback anchor function:DeprecationWarning:mkdocstrings
  ignore:.*v1.*`get_options` method:DeprecationWarning:mkdocstrings
  ignore:.*`config_file_path` argument:DeprecationWarning:mkdocstrings
  ignore:.*from 'mkdocs_autorefs.:DeprecationWarning:mkdocstrings_handlers.python
