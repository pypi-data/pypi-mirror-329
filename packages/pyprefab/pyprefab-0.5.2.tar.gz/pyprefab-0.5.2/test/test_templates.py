"""Snapshot tests for pyprefab templates."""

import pytest

# tomllib is not part of the standard library until Python 3.11
try:
    import tomllib  # type: ignore
except ModuleNotFoundError:
    import tomli as tomllib  # type: ignore


def test_changelog(cli_output, snapshot):
    """Location and contents of CHANGELOG.md are correct."""
    project_path, cli_result = cli_output
    with open(project_path / 'CHANGELOG.md', 'r', encoding='utf-8') as f:
        changelog = f.read()
    assert changelog == snapshot


def test_contributing(cli_output, snapshot):
    """Location and contents of CHANGELOG.md are correct."""
    project_path, cli_result = cli_output
    with open(project_path / 'CONTRIBUTING.md', 'r', encoding='utf-8') as f:
        contributing = f.read()
    assert contributing == snapshot


@pytest.mark.parametrize(
    'docs_file',
    [
        'conf.py',
        'index.rst',
        'CHANGELOG.md',
        'CONTRIBUTING.md',
        'README.md',
        'usage.md',
    ],
)
def test_docs_dir(cli_output, snapshot, docs_file):
    """Documentation contents are correct."""
    project_path, cli_result = cli_output
    with open(project_path / 'docs' / 'source' / docs_file, 'r', encoding='utf-8') as f:
        file = f.read()
    assert file == snapshot


def test_readme(cli_output, snapshot):
    """README.md contents are correct."""
    project_path, cli_result = cli_output
    with open(project_path / 'README.md', 'r', encoding='utf-8') as f:
        readme = f.read()
    assert readme == snapshot


def test_pyproject_docs(cli_output, snapshot):
    """pyproject.toml contents are correct for project with docs."""
    project_path, cli_result = cli_output
    with open(project_path / 'pyproject.toml', 'rb') as f:
        pyproject = tomllib.load(f)
    assert pyproject.get('dependency-groups', {}).get('docs')
    assert pyproject == snapshot


def test_pyproject_no_docs(cli_output_no_docs, snapshot):
    """pyproject.toml contents are correct for project without docs."""
    project_path, cli_result = cli_output_no_docs
    with open(project_path / 'pyproject.toml', 'rb') as f:
        pyproject = tomllib.load(f)
    assert pyproject.get('dependency-groups', {}).get('docs') is None
    assert pyproject == snapshot
