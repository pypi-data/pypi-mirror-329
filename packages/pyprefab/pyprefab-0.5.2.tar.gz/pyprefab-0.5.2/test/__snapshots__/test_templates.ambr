# serializer version: 1
# name: test_changelog
  '''
  # Changelog
  
  All notable changes to transporter_logs are documented here.
  
  The format is based on [Keep a Changelog](https://keepachangelog.com), and the
  project uses [Semantic Versioning](https://semver.org/).
  
  '''
# ---
# name: test_contributing
  '''
  # Contributing to transporter_logs
  
  ## Reporting bugs
  
  If something isn't working as described, or if you find a mistake in the
  documentation, please feel free to report a bug by opening an issue.
  
  ## Contributing to the code base
  
  Contributions to the code base are welcome. If you want to add a new feature,
  please open an issue before doing any work, to ensure that the suggestion
  aligns with the project's goals and overall direction.
  
  If you'd like to tackle an existing issue, please leave a comment on it.
  
  ### Creating your local development environment
  
  For contributing to this code base, you'll need:
  
  - A [GitHub account](https://github.com/)
  - [Git](https://git-scm.com/) installed on your machine
  - **optional**: [uv](https://docs.astral.sh/uv/getting-started/installation/)
  (the Python-based directions below use `uv`, but if you
  already have a preferred Python toolset, that should work too)
  
  > [!IMPORTANT]
  > If you have an active Python virtual environment (for example, conda's
  > base environment), you'll need to deactivate it before following the
  > instructions below.
  
  #### Configure git
  
  1. On GitHub, [fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) this repository.
  
  2. Clone the forked repository to your machine:
  
      ```sh
      git clone https://github.com/<your github username>/<this-repo-name>.git
      cd <this-repo-name>
      ```
  
  3. **optional:** Set the `upstream` remote to sync your fork with this
  repository:
  
      ```sh
      git remote add upstream https://github.com/<this-repo-owner-or-org-name>/<this-repo-name>.git
      git fetch upstream
      ```
  
  #### Install project and dependencies
  
  1. From the root of the repo, create a virtual environment and install the
  project dependencies. The
  [`uv sync` command](https://docs.astral.sh/uv/reference/cli/#uv-sync) handles
  installing Python, creating a virtual environment, and installing project
  dependencies.
  
      ```sh
      uv sync
      ```
  
     (More information about how uv
      [finds or downloads a Python interpreter](https://docs.astral.sh/uv/reference/cli/#uv-python))
  
  2. Run the test suite to check that everything works correctly:
  
      > [!TIP]
      > Prefixing python commands with `uv run` instructs uv to run the command
      > in the project's virtual environment, even if you haven't explicitly
      > activated it.
  
      ```sh
      uv run pytest
      ```
  
  3. Install the `pre-commit` hooks used for linting and other checks (this may
  take a few minutes but only needs to be done once).
  
      ```sh
      uv run pre-commit install
      ```
  
  4. Make sure the `pre-commit` checks are working correctly:
  
      ```sh
      uv run pre-commit install
      ```
  
  ### Updating your development environment
  
  If time has passed between your initial project setup and when you make changes
  to the code, make sure your fork and development environment are up-to-date.
  
  1. Sync your fork to the upstream repository:
  
      ```sh
      git checkout main
      git fetch upstream
      git rebase upstream/main
      git push origin main
      ```
  
  2. Update your project dependencies:
  
      ```sh
      uv sync
      ```
  
  ### Adding project dependencies
  
  If your change requires a new dependency, add it as follows:
  
  ```sh
  uv add <dependency>
  ```
  
  The [`uv add`](https://docs.astral.sh/uv/reference/cli/#uv-add) command will:
  
  - Add the dependency to `pyproject.toml`
  - Install the dependency into the project's virtual environment
  - Update the project's lockfile (`uv.lock`)
  
  Make sure to commit the updated versions of `pyproject.toml` and `uv.lock`.
  
  ### Submitting code changes
  
  After you've completed the changes described in the issue you're working on,
  you can submit them by
  [creating a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork)
  (PR) to this repository.
  
  Please ensure the following are true before creating the PR:
  
  - Your change is covered by tests, if applicable
  - Project documentation is updated, if applicable
  - All tests pass (`uv run pytest`)
  - All pre-commit checks are successful
  (these checks will run automatically as you make commits)
  - The `[Unreleased]` section of [CHANGELOG.md](CHANGELOG.md) contains a
  description of your change.
  
  The PR itself should:
  
  - Have a descriptive title
  - Be [linked to its corresponding issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue)
  in the description.
  - Have a description that includes any other information or context that will
  help a code reviewer understand your changes.
  
  '''
# ---
# name: test_docs_dir[CHANGELOG.md]
  '''
  ```{include} ../../CHANGELOG.md
  ```
  
  '''
# ---
# name: test_docs_dir[CONTRIBUTING.md]
  '''
  ```{include} ../../CONTRIBUTING.md
  ```
  
  '''
# ---
# name: test_docs_dir[README.md]
  '''
  ```{include} ../../README.md
  ```
  
  '''
# ---
# name: test_docs_dir[conf.py]
  '''
  # Configuration file for the Sphinx documentation builder.
  #
  # For the full list of built-in configuration values, see the documentation:
  # https://www.sphinx-doc.org/en/master/usage/configuration.html
  
  # -- Project information -----------------------------------------------------
  # https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
  
  import os
  import sys
  from datetime import datetime
  from importlib import metadata
  
  # If extensions (or modules to document with autodoc) are in another directory,
  # add these directories to sys.path here. If the directory is relative to the
  # documentation root, use os.path.abspath to make it absolute, like shown here.
  sys.path.insert(0, os.path.abspath('...'))
  sys.path.insert(0, os.path.abspath('..'))
  sys.path.insert(0, os.path.abspath('.'))
  
  # -- General configuration ---------------------------------------------------
  # https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
  
  extensions = [
      # auto generate API documents from Python docstrings
      'sphinx.ext.autodoc',
      # used when publishing to GitHub pages: turns off GitHub's jekyll default
      'sphinx.ext.githubpages',
      # allow cross-linking to other project's sphinx-based docs
      'sphinx.ext.intersphinx',
      # for use with autodoc: adds support for Google and NumPy style docstrings
      'sphinx.ext.napoleon',
      # links to Python source code from documentation
      'sphinx.ext.viewcode',
      # adds a copy button to code blocks
      'sphinx_copybutton',
      # generate open graph metadata for docs pages
      'sphinxext.opengraph',
      # use MyST parser for markdown files in Sphinx documentation
      'myst_parser',
  ]
  
  # extensions for the MyST markdown parser
  # https://myst-parser.readthedocs.io/en/latest/syntax/optional.html
  myst_enable_extensions = [
      'colon_fence',
      'fieldlist',
      'replacements',
      'tasklist',
  ]
  
  # Settings from the copybutton's config
  # https://github.com/executablebooks/sphinx-copybutton/blob/master/docs/conf.py
  # There are useful for ensuring that the copy button only copies code
  # (as prefixed by ">>>" for example) and not the prompt or the output.
  copybutton_prompt_text = r'>>> |\.\.\. |\$ |In \[\d*\]: | {2,5}\.\.\.: | {5,8}: '
  copybutton_prompt_is_regexp = True
  copybutton_line_continuation_character = '\\'
  copybutton_here_doc_delimiter = 'EOT'
  copybutton_selector = 'div:not(.no-copybutton) > div.highlight > pre'
  templates_path = ['_templates']
  exclude_patterns = []
  
  # The suffix(es) of source filenames.
  # You can specify multiple suffix as a list of string:
  source_suffix = ['.rst', '.md']
  
  # The encoding of source files.
  # source_encoding = 'utf-8-sig'
  
  # The master toctree document.
  master_doc = 'index'
  
  # project information
  now = datetime.now()
  project = 'transporter_logs'
  copyright = f'{now.year}, Miles O&#39;Brien | Last update {now.strftime("%B %d, %Y")}'
  author = 'Miles O&#39;Brien'
  
  version = metadata.version('transporter_logs')
  release = version
  
  # The language for content autogenerated by Sphinx. Refer to documentation
  # for a list of supported languages.
  language = 'en'
  
  # List of patterns, relative to source directory, that match files and
  # directories to ignore when looking for source files.
  exclude_patterns = ['_build']
  
  # The name of the Pygments (syntax highlighting) style to use.
  pygments_style = 'monokai'
  
  # If true, `todo` and `todoList` produce output, else they produce nothing.
  todo_include_todos = False
  
  # -- Options for HTML output -------------------------------------------------
  # https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
  
  html_theme = 'alabaster'
  html_static_path = ['_static']
  
  # Theme options are theme-specific and customize the look and feel of a theme
  # further. For a list of options available for each theme, see the
  # documentation.
  html_theme_options = {
      'description': f'An app for parsin&#39; &#34;transporter logs&#34;\n{release}',
      # 'github_user': '',
      # 'github_repo': '',
      # 'github_banner': True,
      # 'github_button': False,
      'font_family': "'Roboto', Georgia, sans",
      'head_font_family': "'Roboto', Georgia, serif",
      'code_font_family': "'Roboto Mono', 'Consolas', monospace",
      'pre_bg': '#433e56',
  }
  
  # Add any paths that contain custom themes here, relative to this directory.
  # html_theme_path = []
  
  # The name for this set of Sphinx documents.  If None, it defaults to
  # "<project> v<release> documentation".
  # html_title = None
  
  # A shorter title for the navigation bar.  Default is the same as html_title.
  # html_short_title = None
  
  # The name of an image file (relative to this directory) to place at the top
  # of the sidebar.
  # html_logo = None
  
  # The name of an image file (within the static path) to use as favicon of the
  # docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
  # pixels large.
  # html_favicon = None
  
  # Add any paths that contain custom static files (such as style sheets) here,
  # relative to this directory. They are copied after the builtin static files,
  # so a file named "default.css" will overwrite the builtin "default.css".
  html_static_path = ['_static']
  
  # These paths are either relative to html_static_path
  # or fully qualified paths (eg. https://...)
  html_css_files = [
      'custom.css',
  ]
  
  # Add any extra paths that contain custom files (such as robots.txt or
  # .htaccess) here, relative to this directory. These files are copied
  # directly to the root of the documentation.
  # html_extra_path = []
  
  # If not '', a 'Last updated on:' timestamp is inserted at every page bottom,
  # using the given strftime format.
  html_last_updated_fmt = '%b %d, %Y'
  
  # If true, SmartyPants will be used to convert quotes and dashes to
  # typographically correct entities.
  # html_use_smartypants = True
  
  # Custom sidebar templates, maps document names to template names.
  # html_sidebars = {
  #     '**': [
  #         'about.html',
  #         'navigation.html',
  #         'relations.html',
  #         'searchbox.html',
  #     ]
  # }
  
  # Additional templates that should be rendered to pages, maps page names to
  # template names.
  # html_additional_pages = {}
  
  # If false, no module index is generated.
  # html_domain_indices = True
  
  # If false, no index is generated.
  # html_use_index = True
  
  # If true, the index is split into individual pages for each letter.
  # html_split_index = False
  
  # If true, links to the reST sources are added to the pages.
  # html_show_sourcelink = True
  
  # If true, "Created using Sphinx" is shown in the HTML footer. Default is True.
  html_show_sphinx = False
  
  # If true, "(C) Copyright ..." is shown in the HTML footer. Default is True.
  # html_show_copyright = True
  
  # If true, an OpenSearch description file will be output, and all pages will
  # contain a <link> tag referring to it.  The value of this option must be the
  # base URL from which the finished HTML is served.
  # html_use_opensearch = ''
  
  # This is the file name suffix for HTML files (e.g. ".xhtml").
  # html_file_suffix = None
  
  # Language to be used for generating the HTML full-text search index.
  # Sphinx supports the following languages:
  #   'da', 'de', 'en', 'es', 'fi', 'fr', 'hu', 'it', 'ja'
  #   'nl', 'no', 'pt', 'ro', 'ru', 'sv', 'tr'
  # html_search_language = 'en'
  
  # A dictionary with options for the search language support, empty by default.
  # Now only 'ja' uses this config value
  # html_search_options = {'type': 'default'}
  
  # Output file base name for HTML help builder.
  # htmlhelp_basename = ""
  
  '''
# ---
# name: test_docs_dir[index.rst]
  '''
  .. transporter_logs documentation index file,
     You can adapt this file completely to your liking, but it should at least
     contain the root `toctree` directive.
  
  .. toctree::
     :hidden:
     :maxdepth: 2
     :caption: Contents:
  
     usage
     CONTRIBUTING
     CHANGELOG
  
  .. include:: readme.md
     :parser: myst_parser.sphinx_
  '''
# ---
# name: test_docs_dir[usage.md]
  '''
  # Using transporter_logs
  
  This page contains information about using transporter_logs.
  
  '''
# ---
# name: test_pyproject_docs
  dict({
    'build-system': dict({
      'build-backend': 'setuptools.build_meta',
      'requires': list([
        'setuptools>=45',
        'wheel',
        'setuptools_scm>=8',
      ]),
    }),
    'dependency-groups': dict({
      'dev': list([
        'coverage',
        'pre-commit',
        'pytest',
        'pytest-random-order>=1.1.1',
        'ruff',
      ]),
      'docs': list([
        'myst-parser>=3.0.1',
        'sphinx>=7.4.7',
        'sphinx-autobuild>=2024.10.3',
        'sphinx-copybutton>=0.5.2',
        'sphinxext-opengraph>=0.9.1',
      ]),
    }),
    'project': dict({
      'authors': list([
        dict({
          'name': "Miles O'Brien",
        }),
      ]),
      'dependencies': list([
        'structlog',
      ]),
      'description': 'An app for parsin\' "transporter logs"',
      'dynamic': list([
        'version',
      ]),
      'license': dict({
        'text': 'MIT License',
      }),
      'maintainers': list([
        dict({
          'name': "Miles O'Brien",
        }),
      ]),
      'name': 'transporter_logs',
      'readme': 'README.md',
      'requires-python': '>=3.9',
      'scripts': dict({
        'transporter_logs': 'transporter_logs.app:main',
      }),
      'urls': dict({
      }),
    }),
    'tool': dict({
      'pytest': dict({
        'ini_options': dict({
          'addopts': '--random-order',
          'testpaths': list([
            'test',
          ]),
        }),
      }),
      'ruff': dict({
        'format': dict({
          'quote-style': 'single',
        }),
        'line-length': 120,
        'lint': dict({
          'extend-select': list([
            'I',
            'Q',
          ]),
          'flake8-quotes': dict({
            'inline-quotes': 'single',
          }),
        }),
      }),
      'setuptools': dict({
        'packages': dict({
          'find': dict({
            'namespaces': True,
            'where': list([
              'src',
            ]),
          }),
        }),
      }),
      'setuptools_scm': dict({
        'fallback_version': '0.0.0.dev0',
        'local_scheme': 'no-local-version',
      }),
    }),
  })
# ---
# name: test_pyproject_no_docs
  dict({
    'build-system': dict({
      'build-backend': 'setuptools.build_meta',
      'requires': list([
        'setuptools>=45',
        'wheel',
        'setuptools_scm>=8',
      ]),
    }),
    'dependency-groups': dict({
      'dev': list([
        'coverage',
        'pre-commit',
        'pytest',
        'pytest-random-order>=1.1.1',
        'ruff',
      ]),
    }),
    'project': dict({
      'authors': list([
        dict({
          'name': "Miles O'Brien",
        }),
      ]),
      'dependencies': list([
        'structlog',
      ]),
      'description': 'An app for parsin\' "transporter logs"',
      'dynamic': list([
        'version',
      ]),
      'license': dict({
        'text': 'MIT License',
      }),
      'maintainers': list([
        dict({
          'name': "Miles O'Brien",
        }),
      ]),
      'name': 'transporter_logs',
      'readme': 'README.md',
      'requires-python': '>=3.9',
      'scripts': dict({
        'transporter_logs': 'transporter_logs.app:main',
      }),
      'urls': dict({
      }),
    }),
    'tool': dict({
      'pytest': dict({
        'ini_options': dict({
          'addopts': '--random-order',
          'testpaths': list([
            'test',
          ]),
        }),
      }),
      'ruff': dict({
        'format': dict({
          'quote-style': 'single',
        }),
        'line-length': 120,
        'lint': dict({
          'extend-select': list([
            'I',
            'Q',
          ]),
          'flake8-quotes': dict({
            'inline-quotes': 'single',
          }),
        }),
      }),
      'setuptools': dict({
        'packages': dict({
          'find': dict({
            'namespaces': True,
            'where': list([
              'src',
            ]),
          }),
        }),
      }),
      'setuptools_scm': dict({
        'fallback_version': '0.0.0.dev0',
        'local_scheme': 'no-local-version',
      }),
    }),
  })
# ---
# name: test_readme
  '''
  # transporter_logs
  
  An app for parsin' "transporter logs"
  
  '''
# ---
